react/week4/my-app/src/Context.js

```diff
@@ -0,0 +1,82 @@
+import React, { useState, useEffect } from "react";
+
+const Context = React.createContext();
+
+function ContextProvider({ children }) {
+  const [userInput, setUserInput] = useState("");
+  const [data, setData] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(null);
+  useEffect(() => {
+    if (!userInput || error) {
+      setLoading(false);
+    } else {
+      setLoading(true);
+      fetchGitUsers(userInput);
+    }
+  }, [userInput, error]);
+  const fetchGitUsers = async (input) => {
+    try {
+      await fetch(`https://api.github.com/search/users?q=${input}`)
+        .then((response) => {
+          if (!response.ok) {
+            throw Error("Could not fetch the data ");
+          } else {
+            return response.json();
+          }
+        })
+        .then((data) => {
+          setData(data.items);
+          //console.log(data);
+          setLoading(false);
+        });
+    } catch (error) {
+      setError(error.message);
+    }
+  };
+
+  return (
+    <Context.Provider
+      value={{
+        userInput,
+        setUserInput,
+        data,
+        setData,
+        loading,
+        setLoading,
+        error,
+        setError,
+      }}
+    >
+      {/* <pre>{JSON.stringify(data, null, 0)}</pre> */}
+      {children}
+    </Context.Provider>
+  );
+}
+
+export const useUser = () => React.useContext(Context);
```

Comment:
Did you see this approach from some other place? The approach works fine but just for the sake of this homework, could you instead export the context and the use `useContext` in the components that need access to the context?

Reply:
yes I did this the way you showed in online class but for some reason didn't work so I asked from one of the classmate so we change it to 😅